dashboard/app/controllers/potential_teachers_controller.rb

```diff
@@ -33,7 +33,7 @@ def show
 
   def send_hoc_email(params)
     name = current_user&.name || params[:name]
-    email = current_user&.email || params[:email]
+    email = current_user&.email.present? ? current_user.email : params[:email]
```

Comment:
@carl-codeorg pointed out that this was returning the empty string, so I updated to check first if the email is present